### ISSUE TITLE
Tree Shaking Support

### ISSUE THREAD

**Issue by schlosser** [garmeeh/next-seo#663]
Hello! I'd like to import `NextSEO` into my project without including all of the JSON LD functions I'm not using. At the moment `next-seo` is taking up a significant amount of my bundle size for all pages.

I tried following @joeyorlando's steps here: https://github.com/garmeeh/next-seo/issues/75#issuecomment-705152810, to no avail. Is there a supported way to implement tree shaking?

**Comment by schlosser** [comment]
Yep! I'd love a response to understand if anyone has had any luck tree shacking the JSON LD code from `next-seo`.

**Comment by iffa** [comment]
Couldn't get any of the steps working either, so +1

**Comment by schlosser** [comment]
Definitely still relevant!

**Comment by riccardogiorato** [comment]
Definitely still relevant!

**Comment by mryechkin** [comment]
Still relevant 💯

**Comment by riccardogiorato** [comment]
Definitely still relevant!

**Comment by garmeeh** [comment]
Yes apologies @schlosser. I need to disable stale bot 🤦‍♂️

**Comment by mryechkin** [comment]
Still relevant!

**Comment by kuus** [comment]
hi @garmeeh , thanks for this great library,
have you tried already setting `"sideEffects": false,` in the `package.json`?
assuming there are no sideEffects that would help I guess, otherwise you can list those files which do have sideEffects in an array,
see [webpack](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free) docs

cheers

**Comment by dkelsen** [comment]
I'm also interested in tree shaking support.

**Comment by icazevedo** [comment]
Yeah, same! It'd be really useful to have tree shaking support.
